dnf/rpm/miscutils.py

```diff
@@ -13,47 +13,77 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 # Copyright 2003 Duke University
 
-from __future__ import print_function, absolute_import
-from __future__ import unicode_literals
+from __future__ import print_function, absolute_import, unicode_literals
 
-import rpm
 import os
 import subprocess
 import logging
-
-from dnf.i18n import ucd
-from dnf.i18n import _
 from shutil import which
 
+from dnf.i18n import _
 
-logger = logging.getLogger('dnf')
-
+_logger = logging.getLogger('dnf')
+_rpmkeys_binary = which("rpmkeys")
+_logger.debug(_('Using rpmkeys executable from %s to verify signatures'),
```

Comment:
Unfortunately I found out this doesn't log anything, I think it's because at the time this runs the `dnf` logger has not been set up yet.

I tried to figure out how we could workaround this and the best suggestion I have is setting the global `_rpmkeys_binary = None` and them using something like:
```python
def _find_rpmkeys_binary():
    global _rpmkeys_binary
    if not _rpmkeys_binary:
        _rpmkeys_binary = which("rpmkeys")
        _logger.info(_('Using rpmkeys executable from %s to verify signatures'), _rpmkeys_binary)
```

Reply:
Is DNF multi-threaded?  This kind of code makes me nervous.  If you can promise that DNF is and will always be single-threaded, then this is fine.

Reply:
Yes I believe that is safe for this python implementation.